cluster/eval_server/add_model.py

```diff
@@ -0,0 +1,155 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+sys.path.insert(0, '.')
+
+import datetime
+import hashlib
+import os
+import shutil
+
+from absl import flags
+from google.cloud import bigtable
+from tensorflow import gfile
+import fire
+
+from bigtable_input import METADATA
+
+
+flags.mark_flags_as_required([
+    "cbt_project", "cbt_instance", "cbt_table"
+])
+
+FLAGS = flags.FLAGS
+
+
+# TODO(sethtroisi): Unify with oneoffs
+MODEL_ROW = "m_{}_{:0>10}"
+MODEL_NAME = b"model"
+
+PLAYER_FOLDER = "gs://minigo-pub/eval_server/models/"
+
+GTP_BIN = "bazel-bin/cc/gtp"
+LIB_TF_FW_SO = "cc/tensorflow/lib/libtensorflow_framework.so"
+LIB_TF_CC_SO = "cc/tensorflow/lib/libtensorflow_cc.so"
+
+
+def verify_params(model_path, flags, binary_path):
+    # Verify model_path = .../<model>.pb
+    assert model_path.endswith('.pb'), model_path
+
+    assert flags, "Flags must be non-empty"
+    assert "-num_readouts=" in flags, "must set playouts"
+    assert "-model" not in flags, "leave model unset"
+
+    # Verify binary_path is to a gtp binary and it exists
+    binary_base = binary_path.replace(GTP_BIN, "")
+    assert binary_path.endswith(GTP_BIN), binary_path
+    assert os.path.isfile(binary_path)
+    # $ ldd bazel-bin/cc/gtp
+    #    libtensorflow_framework.so => (bazel path) => cc/tensorflow/lib/libtensorflow_framework.so
+	#    libtensorflow_cc.so => (bazel path) => cc/tensorflow/lib/libtensorflow_cc.so
+    #    libcuda.so.1
+    #    libnvidia-fatbinaryloader.so.<version e.g. 390>.116
+    assert os.path.isfile(os.path.join(binary_base, LIB_TF_FW_SO))
+    assert os.path.isfile(os.path.join(binary_base, LIB_TF_CC_SO))
+
+    return True
+
+def copy_to_gcs(src, dst):
+    assert gfile.Exists(src)
+    assert not gfile.Exists(dst)
+
+    with gfile.GFile(src, "rb") as src_f, gfile.GFile(dst, "wb") as dst_f:
+        shutil.copyfileobj(src_f, dst_f)
+
+
+def add_model(name, model_path, flags, binary_path):
+    """Add a player (model + flags) to models cbt
+    name: name of the model
```

Comment:
what is the name here, actually?  e.g. "000990-cormorant", or v17-990, or some other scheme?  This is a good place to take a no-op comment ("name: the name") and give some clarity.

Reply:
it's a free form name field. I'll add an assert it conforms to [a-Z0-9-._]

Reply:
ok.  I'm saying this is the chance to describe how the caller should use it.  I.e. how you think the names should be used or what makes a good name or an example...